From 3a7c9ce5c5685ae644f2b535a1ec6fc52b12171d Mon Sep 17 00:00:00 2001
From: Cesar Talledo <cesar.talledo@docker.com>
Date: Thu, 8 May 2025 20:44:46 -0700
Subject: [PATCH] Add a crio-static target using local build (much faster than
 the build-static target).

Signed-off-by: Cesar Talledo <cesar.talledo@docker.com>
---
 Makefile | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index e2f0c018d4e..070bf35437b 100644
--- a/Makefile
+++ b/Makefile
@@ -99,6 +99,7 @@ BASE_LDFLAGS = ${SHRINKFLAGS} \
 	-X ${PROJECT}/internal/version.buildDate=${BUILD_DATE}
 
 GO_LDFLAGS = -ldflags '${BASE_LDFLAGS} ${EXTRA_LDFLAGS}'
+GO_LDFLAGS_STATIC = -ldflags '${BASE_LDFLAGS} ${EXTRA_LDFLAGS} -w -extldflags -static'
 
 define curl_to
 	curl -sSfL --retry 5 --retry-delay 3 "$(1)" -o $(2)
@@ -176,7 +177,7 @@ $(SHELLCHECK): $(BUILD_BIN_PATH)
 ##@ Build targets:
 
 .PHONY: binaries
-binaries: bin/crio bin/pinns ## Build all binaries.
+binaries: bin/crio bin/pinns bin/crio-static ## Build all binaries.
 
 .PHONY: test-binaries
 test-binaries: ## Build all test-binaries.
@@ -204,6 +205,9 @@ test/nri/nri.test: $(wildcard test/nri/*.go) ## Build the NRI test binary.
 bin/crio: $(GO_FILES) ## Build the CRI-O main binary.
 	$(GO_BUILD) $(GCFLAGS) $(GO_LDFLAGS) -tags "$(BUILDTAGS)" $(GO_BUILDFLAGS) -o $@ ./cmd/crio
 
+bin/crio-static: $(GO_FILES)
+	$(GO_BUILD) $(GCFLAGS) $(GO_LDFLAGS_STATIC) -tags "$(BUILDTAGS) netgo osusergo static_build containers_image_openpgp" -o $@ ./cmd/crio
+
 .PHONY: build-static
 build-static: ## Build the static binaries.
 	$(CONTAINER_RUNTIME) run --network=host --rm --privileged -ti -v /:/mnt \
